frontend/src/page/Realm.tsx

```diff
@@ -1,43 +1,57 @@
 import React from "react";
 import { Link } from "react-router-dom";
 
+import { graphql, useLazyLoadQuery } from "react-relay/hooks";
+import { RealmQuery } from "../query-types/RealmQuery.graphql";
+
 import { Breadcrumbs } from "../ui/Breadcrumbs";
 
 
 type Props = {
-    path: string[];
+    path: string;
 };
 
 export const Realm: React.FC<Props> = ({ path }) => {
-    const isRoot = path.length === 0;
-    const ids = resolvePath(path);
-    if (ids == null) {
+    const isRoot = path === "";
+
+    // TODO Build this query from fragments!
+    const { realm } = useLazyLoadQuery<RealmQuery>(graphql`
+        query RealmQuery($path: String!) {
+            realm: realmByPath(path: $path) {
+                name
+                parents { name path }
+                children { id name path }
+            }
+        }
+    `, {
+        // The API expects a "leading" slash to non-root paths to separate
+        // the first component from the root path segment `''`
+        path: isRoot ? "" : `/${path}`,
+    });
+
+    if (!realm) {
         // TODO: that should obviously be handled in a better way
-        return <b>Realm path not found :(</b>;
+        //   Also: I'm not sure whether that's still the only cause for error.
+        return <b>{"Realm path not found :("}</b>;
     }
 
-    const realmId = ids[ids.length - 1];
-    const realm = REALMS[realmId];
-
     // Prepare data for breadcrumbs
     const breadcrumbs = [];
-    let tmpPath = "/r";
-    for (const id of ids.slice(1)) {
-        tmpPath += `/${REALMS[id].path}`;
+    for (const { name, path } of realm.parents.slice(1)) {
         breadcrumbs.push({
-            label: REALMS[id].name,
-            href: tmpPath,
+            label: name,
+            href: `/r${path}`,
         });
     }
```

Comment:
Mhhhh couldn't this be `realm.parents.slice(1).map(({ name, path }) => ({ ... }))`?

Reply:
Yesss! :grimacing: